tests/platform_helper/domain/test_codebase.py

```diff
@@ -0,0 +1,620 @@
+import filecmp
+import json
+import os
+from datetime import datetime
+from pathlib import Path
+from unittest.mock import MagicMock
+from unittest.mock import Mock
+from unittest.mock import PropertyMock
+from unittest.mock import call
+from unittest.mock import patch
+
+import boto3
+import click
+import pytest
+import requests
+
+from dbt_platform_helper.domain.codebase import Codebase
+from dbt_platform_helper.exceptions import AWSException
+from dbt_platform_helper.utils.application import ApplicationNotFoundError
+from dbt_platform_helper.utils.application import Environment
+from tests.platform_helper.conftest import EXPECTED_FILES_DIR
+
+real_ecr_client = boto3.client("ecr")
+real_ssm_client = boto3.client("ssm")
+
+
+def mock_aws_client(get_aws_session_or_abort):
+    session = MagicMock(name="session-mock")
+    client = MagicMock(name="client-mock")
+    session.client.return_value = client
+    get_aws_session_or_abort.return_value = session
+    return client
+
+
+class CodebaseMocks:
+    def __init__(self):
+        self.load_application_fn = Mock()
+        self.get_aws_session_or_abort_fn = Mock()
+        self.input_fn = Mock(return_value="yes")
+        self.echo_fn = Mock()
+        self.confirm_fn = Mock(return_value=True)
+        self.check_codebase_exists_fn = Mock(
+            return_value="""
+                                             {
+                                                "name": "test-app", 
+                                                "repository": "uktrade/test-app",
+                                                "services": "1234"
+                                             }
+                                        """
+        )
+        self.check_image_exists_fn = Mock(return_value="")
+        self.subprocess = Mock()
+
+    def params(self):
+        return {
+            "load_application_fn": self.load_application_fn,
+            "get_aws_session_or_abort_fn": self.get_aws_session_or_abort_fn,
+            "check_codebase_exists_fn": self.check_codebase_exists_fn,
+            "check_image_exists_fn": self.check_image_exists_fn,
+            "input_fn": self.input_fn,
+            "echo_fn": self.echo_fn,
+            "confirm_fn": self.confirm_fn,
+            "subprocess": self.subprocess,
+        }
+
+
+@patch("requests.get")
```

Comment:
maybe overkill but is it possible to also inject requests like subprocess meaning you can mock this instead of patching?

Reply:
we only use it one place at the moment and it's a standard library so probably not great to add yet another 'injectable' argument to the class